### DISCUSSION TITLE
mpc-hc调用mpc-be的视频解码器实现更多格式的硬解

### DISCUSSION THREAD

**Discussion by hooke007** [hooke007/mpv_PlayKit#41] [其它播放器]
## 理由
- be的解码器开发更激进（LAVFilters 太保守）
- 支持N卡的nvdec
- 没有去除4k120fps的硬解码支持
- 折腾

## 操作流程

1. 下载be 1.6.0 以上的版本提供的独立滤镜包解压到任意地方
https://sourceforge.net/projects/mpcbe/files/MPC-BE/Release%20builds/
`standalone_filters-mpc-be.1.6.?.????.x64`（实际上你要用到的只是其中的 `MPCVideoDec.ax` ）

2. 进入mpc选项设置中的扩展滤镜，手动导入刚刚的 `MPCVideoDec.ax`
导入后列表中会出现如图两个新滤镜，把 `MPC Video Decoder` 设为首选  
<img src="https://user-images.githubusercontent.com/41094733/140768116-bcf8f7bc-c8f5-4807-a5b1-4655e03b6551.jpg" width=500>

3. 双击 `MPC Video Decoder` 修改解码为 `nvdec`
<img src="https://user-images.githubusercontent.com/41094733/140768207-731846f3-f4de-4390-a669-a05c5e4de485.jpg" width=500>

## 效果预览

nvdec硬解 h265 444p10

<img src="https://user-images.githubusercontent.com/41094733/140768442-2c5f2c80-7342-4a32-8dad-c663543dfb53.jpg" width=800>

amd硬解 h264 420p8 120fps

<img src="https://user-images.githubusercontent.com/41094733/140768476-ce11542a-13b9-463f-9729-bb1b550b9b10.jpg" width=800>

**Comment by hingbong** [comment]
这个硬解对比现在这个MPV，支持的编码更多吗

**Reply by hooke007** [reply]
不会，这是对比的mpc-hc自带的lavfilter不是mpv
